### ISSUE TITLE
Confusing error when remote path is wrong

### ISSUE THREAD

**Issue by birdsarah** [dask/dask#4741]
Here's an error I recently received when running `df = dd.read_parquet('s3://birdsarah-data-store-live/javascript_10percen_value_1000_only.parquet')`

```
~/Dev/birdsarah/dask/dask/dataframe/io/parquet.py in _read_pyarrow(fs, fs_token, paths, columns, filters, categories, index, infer_divisions)
    678 
    679     dataset = pq.ParquetDataset(paths, filesystem=get_pyarrow_filesystem(fs),
--> 680                                 filters=filters)
    681     if dataset.partitions is not None:
    682         partitions = [n for n in dataset.partitions.partition_names

~/miniconda3/envs/ovscrptd/lib/python3.7/site-packages/pyarrow/parquet.py in __init__(self, path_or_paths, filesystem, schema, metadata, split_row_groups, validate_schema, filters, metadata_nthreads)
    888          self.common_metadata_path,
    889          self.metadata_path) = _make_manifest(
--> 890             path_or_paths, self.fs, metadata_nthreads=metadata_nthreads)
    891 
    892         if self.common_metadata_path is not None:

~/miniconda3/envs/ovscrptd/lib/python3.7/site-packages/pyarrow/parquet.py in _make_manifest(path_or_paths, fs, pathsep, metadata_nthreads)
   1063             if not fs.isfile(path):
   1064                 raise IOError('Passed non-file path: {0}'
-> 1065                               .format(path))
   1066             piece = ParquetDatasetPiece(path)
   1067             pieces.append(piece)

OSError: Passed non-file path: birdsarah-data-store-live/javascript_10percen_value_1000_only.parquet
```

The path I passed in had a typo in it, and when I correct it everything is fine. However, I spent a long time on this because by the time the error got kicked back to me the `s3://` prefix has been stripped and I though it was looking for the file on my local system and that I was working with s3 wrong in some way.

I know that error is coming from `pyarrow`, but I thought I would open it here first incase it was best handled by catching the error in `dataframe/io/parquet.py`.

**Comment by martindurant** [comment]
I could imagine catching this and providing a better message, but it would depend on knowing what error exactly pyarrow would raise - which seems like an implementation deal. I would prefer for pyarrow to raise a more useful/correct error. They are currently working on parquet/dask related things, so this would be a good time to raise it with them.

**Comment by birdsarah** [comment]
Agree. I guess it depends on what dask sends to pyarrow as to whether they can "do the right thing". Does dask pre-separate the path and initiate s3 access. If yes, then I would guess dask will need to stitch it back together to error report well. If no, then 100% will raise with pyarrow.

**Comment by martindurant** [comment]
I think the protocol is stripped, but pyarrow does know the file-system that is in use (so it can make a reasonable message out of that)

**Comment by quasiben** [comment]
@birdsarah did you end up filing an issue with pyarrow folks ?

**Comment by birdsarah** [comment]
Not yet. I wanted to dig into the code path a bit more to see what gets passed from dask (which i'm not going to have time to do soon)

**Comment by GenevieveBuckley** [comment]
Closing this - @birdsarah if you do end up making an issue over at pyarrow, please add a link here so that interested people can follow any progress updates.
